fuzzers/main.py

```diff
@@ -0,0 +1,450 @@
+    i.add_watch((local / path).as_posix())
+    while True:
+        # at most every 5 seconds check for exit_event
+        for event in i.event_gen(yield_nones = False, timeout_s = 5):
+            (_, event_types, _, filename) = event
+            local_filename = local / path / filename
+            remote_filename = f'{remote}/{path}/{filename}'
+            if 'IN_CLOSE_WRITE' in event_types:
+                log_file.write(
+                    f"Uploading new corpus item {local_filename} to GCS {remote_filename}\n"
+                )
+                log_file.flush()
+                try:
+                    # TODO: batch uploads
+                    bucket.blob(remote_filename).upload_from_filename(local_filename)
+                    on_uploaded_item(remote_filename)
+                except FileNotFoundError:
+                    pass # Ignore, as it'd mean the file has been deleted already
+            if 'IN_DELETE' in event_types and not ignore_deletions:
+                log_file.write(
+                    f"Removing now-removed corpus item {local_filename} as GCS {remote_filename}\n"
+                )
+                log_file.flush()
+                try:
+                    # TODO: batch
+                    bucket.blob(remote_filename).delete()
+                    on_deleted_item(remote_filename)
+                except google.api_core.exceptions.NotFound:
+                    pass # Ignore, as it'd mean the file isn't there already
+            if exit_event.is_set():
+                return
+        if exit_event.is_set():
+            return
+
+def auto_upload_to_gcs(
+    *,
+    exit_event: threading.Event,
+    path: pathlib.Path,
+    local: pathlib.Path,
+    remote: str,
+    bucket: str,
+    log_file: typing.IO[str],
+    ignore_deletions: bool,
+    # Without Any here mypy doesn't let us use lambdas. The result is ignored anyway.
+    on_uploaded_item: typing.Callable[[str], typing.Any],
+    on_deleted_item: typing.Callable[[str], typing.Any],
+) -> None:
+    """
+    Setup an inotify thread that will automatically upload changes to GCS
+
+    `exit_event` will stop the thread when set
+    The contents of `local/path` will be propagated to `remote/path` in `bucket`
+    Logs will be sent to `log_file`.
+    If `ignore_deletions` is set, deletions will not be propagated.
+    Upon successful uploading of deletion or an item, resp. `on_uploaded_item` and
+    `on_deleted_item` will be called.
+    """
+    # pylint: disable=line-too-long
+    print(f"Setting up inotify watch to auto-upload changes to {local / path} to GCS {remote}/{path}/")
+    log_file.write(f"Setting up inotify watch to auto-upload changes to {local / path} to GCS {remote}/{path}/\n")
+    log_file.flush()
+    # pylint: enable=line-too-long
+
+    threading.Thread(
+        daemon = True,
+        target = uploader_thread,
+        args = (path, local, remote, bucket, exit_event, log_file),
+    ).start()
```

Comment:
By the way, alternative approach is to have a class which inherits from `threading.Thread` and implements `run` method.